src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -1079,6 +1081,37 @@ public void testCreateDatasetRegistration_invalidFileName() {
     assertEquals(HttpStatusCodes.STATUS_CODE_BAD_REQUEST, response.getStatus());
   }
 
+  @Test
+  public void testGetStudyByIdNoDatasets() {
+    Study study = new Study();
+    study.setStudyId(1);
+    study.setName("asdfasdfasdfasdfasdfasdf");
+    when(datasetService.findStudyById(1)).thenReturn(study);
+    initResource();
+    Response response = resource.getStudyById(1);
+    assertEquals(200, response.getStatus());
+    assertEquals(study, response.getEntity());
+  }
+
+  @Test
+  public void testGetStudyByIdWithDatasets() {
```

Comment:
This currently does not test whether datasets exist because `Set<Integer> datasetIds = study.getDatasetIds();` returns `null` and thus will skip adding the datasets to the study. As a result, these lines are not covered.

To resolve this, mock the response from `study.getDatasetIds();` to return a list of integers and `datasetService.findDatasetsByIds(datasetIdList);` to return a list of dummy datasets.